### PR TITLE
polys: Iplemented the Kapur-Saxena-Yang approach to Dixon's resultant

### PR DESCRIPTION
**Pull request by ctsiagkalis**
Iplemented the Kapur-Saxena-Yang approach to Dixon's resultant

#### Brief description of what is fixed or changed
With this update you can check whether the Kapur-Saxena-Yang precondition
(the column corresponding to monomial 1 = x_1 ^ 0 * x_2 ^ 0 * ... * x_n ^ 0
is not a linear combination of the remaining ones) holds and use
`get_KSY_Dixon_resultant()` even in cases the classical Dixon Resultant
fails to yield information about common zeros.

#### Release Notes

* polys
  * Added `KSY_precondition()` to test for the Kapur-Saxena-Yang precondition
  * Added `get_KSY_Dixon_resultant()` and auxiliary methods for computing the KSY approach to Dixon's Resultant

### CODE REVIEW

File: sympy/polys/multivariate_resultants.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def get_max_degrees(self, polynomial):
 
     def get_dixon_matrix(self, polynomial):
         r"""
-        Construct the Dixon matrix from the coefficients of polynomial
```
Comment (jksuom) [inline sympy/polys/multivariate_resultants.py]: This is actually the form recommended by [PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings) for one-line docstrings and also for the summary lines of [multi-line docstrings](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings).

Comment (ctsiagkalis) [reply sympy/polys/multivariate_resultants.py]: Thanks for the review. Just to be clear, did you mean I should remove the 's' that I added or change the whole docstring according to PEP 257? 
I noticed that none of the docstrings is formatted correctly. Should I change all the docstrings of this class?

Comment (jksuom) [reply sympy/polys/multivariate_resultants.py]: > Should I change all the docstrings of this class?

That is not necessary, but I think that this one could be left as is and new docstrings should preferably conform to the recommendation (blank line after summary line).

Comment (ctsiagkalis) [reply sympy/polys/multivariate_resultants.py]: @jksuom I made some changes. Let me know if you I think there is anything else I should change.

File: sympy/polys/multivariate_resultants.py [hunk 2]
```diff
@@ -185,6 +185,58 @@ def get_dixon_matrix(self, polynomial):
...
+            return False
+
+        m, n = matrix.shape
+
+        # simplify the matrix and keep only its non-zero rows
+        matrix = simplify(matrix.rref()[0])
+        rows = [i for i in range(m) if any(matrix[i, j] != 0 for j in range(n))]
+        matrix = matrix[rows,:]
+
+        condition = Matrix([[0 for i in range(n-1)] + [1]])
```
Comment (jksuom) [inline sympy/polys/multivariate_resultants.py]: I think the argument could be `[0]*(n-1) + [1]`.

**Comment by jksuom** [comment]
I have one comment. Otherwise this looks good.

File: sympy/polys/multivariate_resultants.py (hunk 2 above)
Comment (ctsiagkalis) [reply sympy/polys/multivariate_resultants.py]: Thank you. This looks better.

**Comment by jksuom** [comment]
Thanks. I think this is ready to be merged.

**Comment by ctsiagkalis** [comment]
Thanks for your help and advice.

**Pull request merged**